os/board/rtl8720e/src/component/wifi/driver/include/wifi_constants.h

```diff
@@ -698,7 +698,7 @@ typedef enum {
 	WIFI_EVENT_WPA_EAPOL_RECVD,
 
 	/* csi rx done event */
-	WIFI_EVENT_CSI_DONE,
+	WIFI_EVENT_CSI_DONE = 19,
```

Comment:
Does this not show build warning about duplicated definition? Could you let me know why this is necessary even this is defined in os/include/tinyara/wifi/rtk/wifi_constants.h?

Reply:
in example_wifi_csi.c, the function wifi_reg_event_handler(WIFI_EVENT_CSI_DONE, example_wifi_csi_report_cb, NULL); use WIFI_EVENT_MAX in this file as the max event, so if i don't define it here, the WIFI_EVENT_MAX will be treated as 16, and it will not register this event.

Reply:
Dear @sunghan-chang ,
in example_wifi_csi.c, the function wifi_reg_event_handler(WIFI_EVENT_CSI_DONE, example_wifi_csi_report_cb, NULL); use WIFI_EVENT_MAX in this file as the max event, so if i don't define it here, the WIFI_EVENT_MAX will be treated as 16, and it will not register this event.

Reply:
It means you don't use `os/include/tinyara/wifi/rtk/wifi_constants.h` in the app. That uses this header. Why does that example use internal header, not public header?

Reply:
Dear @sunghan-chang,
Sorry it was my mistake, i should not include **os/include/tinyara/wifi/rtk/wifi_constants.h** in this example. This example is a demonstration on how to use wificsi related API and it is calling our internal driver directly so it uses internal header. I will revert the changes in **os/include/tinyara/wifi/rtk/wifi_constants.h,** and define a hardcoded value **WIFI_EVENT_CSI_DONE**   in **example_wifi_csi.c** to pass into **wifi_reg_event_handler,** .  In this case, i will use the **WIFI_EVENT_CSI_DONE** value defined in our internal header, is this ok?

Reply:
@zhongnuo-tang It depends on your API implementation. When we use CSI APIs, do we need to use that definition? If yes, it should be defined in public header, not in internal, and the example should use the public. If upper layer is no need to use it, then it should be in internal.

Reply:
@zhongnuo-tang Anyway duplicate two same header could cause some missing. I suggest removing one and using the same header at all of codes.

Reply:
Dear @sunghan-chang,  
From my understanding, when registering CSI event, it will use api in 8720e/src/component/wifi/api, which refers to internal header. In order to refer to public header, will need Samsung to come up with the framework first. So is it possible to merge this PR first, then when the framework is ready, then modify the example to use public header, is this okay? Thank you.

Reply:
@zhongnuo-tang Because a framework is a different module and a layer to the Realtek CSI module (BSP), we need any some header to call them. What should we use for that?

Reply:
@zhongnuo-tang Once I check a proper header, I will merge this as it is. But later we could ask you to tidy-up the headers.

Reply:
@sunghan-chang Okay thank you!